### PR TITLE
Buyer should receive an order confirmation

### PR DESCRIPTION
**Pull request by Christabll**
ft/order-confirmation
This  branch is implementing the functionality for the system to generate a unique tracking number for each order upon successful payment and send a confirmation email to the buyer's email address with order details, tracking number, and a link to the order status page on the website

- git clone from repo
- git checkout  ft/order-confirmation
- set .env as shown in .envexample
- run npm i
- run npm run dev
- you have to make sure you have done with payment 
- Then you'll receive email message with tracking number inside
- you'll have link in in your email where you click and get your order details for the payment you have made
- Incase you want to use Postman
-  copy this url in your postman: http://localhost:{port}/api/v1/orderDetails/:trackingId
-that trackingId is the one you'll received in email
-then after you'll get you order details
- you can use also swager documentation same process as long as you got email with tracking number

### CODE REVIEW

File: src/controllers/shoppingCartController/viewing.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  const trackingnumber = req.params.trackingId;
+  const order = await db.Order.findOne({
+    where: {
+        trackingNumber : trackingnumber
+    }
+  })
+  const details = await db.OrderDetails.findAll({
+    where: { id: { [Op.in]: order.products } },
+  });
+  res.status(200).json({data: details})
```
Comment (FabCode67) [inline src/controllers/shoppingCartController/viewing.js]: can you include the status and message properties within response

**Review by BenoitIT — APPROVED** [review]
Things work successfully

**Review by FabCode67 — CHANGES_REQUESTED** [review]

File: src/controllers/shoppingCartController/addToCartController.js [hunk 2]
```diff
@@ -321,3 +330,56 @@ const cart_process = async (req, res) => {
...
+      </tbody></table>
+      <p class="m_73160151937089879size-16" style="Margin-top:0;Margin-bottom:0;font-size:16px;line-height:24px" lang="x-size-16"><strong><span style="color:#000000">You sent a payment of ${amount} to Legends E-commerce </span></strong></p><p style="Margin-top:20px;Margin-bottom:0"><span style="color:#000000">You are in safe hands.</span></p>
+      <p><h4><b>Your tracking number is ${trackingNumber}.<b></h4></p>
+      <p style="Margin-top:10px;Margin-bottom:0"><span style="color:#000000">Here is your order details click on view link below!</span></p>
+      <p style="Margin-top:20px;Margin-bottom:20px"><span style="color:#000000"> link <a href= "http://${process.env.HOST}:5000/api/v1/orderDetails/${trackingNumber}">view</a></span></p>
+      
+      <p style="Margin-top:0px;Margin-bottom:10px"><span style="color:#000000">Thank you for shopping with us</span></p>
+      <div style="margin:0%;background:#fcfcfc;padding:1% 2%">
+      <p><h4>We are honored to gain you as Customer and Hope to serve you long time. </h4></p>
+      </div>`,
```
Comment (FabCode67) [inline src/controllers/shoppingCartController/addToCartController.js:377]: let's have the common email footer template

**Review by FabCode67 — CHANGES_REQUESTED** [review]

File: src/controllers/shoppingCartController/viewing.js (hunk 1 above)
Comment (Christabll) [reply src/controllers/shoppingCartController/viewing.js]: check again

**Review by Evaristekanova — APPROVED** [review]

**Review by ngoma3 — APPROVED** [review]
Good job

**Review by Aimeana100 — COMMENTED** [review]
@Christa22 , you have more than 2 commits!!

**Review by FabCode67 — APPROVED** [review]

File: src/controllers/shoppingCartController/viewing.js (hunk 1 above)
Comment (FabCode67) [reply src/controllers/shoppingCartController/viewing.js]: DONE

**Pull request merged**